### PR TITLE
Changelog and version updates for v1.2.3

### PR DESCRIPTION
**Pull request by jgreat**
### Motivation

- Add script `.internal-ci/util/bump_cargo_version.sh` to parse and bump MobileCoin authored packages in this repo.
- Update Changelog for v1.2.3 release
- Update Cargo.lock, Cargo.toml versions for v1.2.3 release
- Update `android-bindings/lib-wrapper/android-bindings/publish.gradle` version for v1.2.3 release.

### CODE REVIEW

File: .internal-ci/util/bump_cargo_version.sh [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        echo "!! File doesn't have a .packages.authors array - skipping"
+    fi
+done
+
+# bump gradle file
+if [[ -n "${dry_run}" ]]
+then
+    echo "-- Update version in android-bindings/lib-wrapper/android-bindings/publish.gradle"
+else
+    sed -i -e "s/${current_tag}/${new_tag}/g" android-bindings/lib-wrapper/android-bindings/publish.gradle
```
Comment (nick-mobilecoin) [inline .internal-ci/util/bump_cargo_version.sh:120]: Nice to see this!  Gets missed often

File: .internal-ci/util/bump_cargo_version.sh [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+                # I would use dasel here, but the output file ends up with properties in alpha order. this will make a mess :-(
+                # so punt and use sed.
+                if [[ -n "${dry_run}" ]]
+                then
+                    echo "sed -i -e \"s/version = \\\"${current_tag}\\\"/version = \\\"${new_tag}\\\"/g\" ${f}"
+                else
+                    sed -i -e "s/version = \"${current_tag}\"/version = \"${new_tag}\"/g" "${f}"
+                fi
+            else
+                echo "!! File current version ${p_version} doesn't match --current-tag ${current_tag}"
```
Comment (nick-mobilecoin) [inline .internal-ci/util/bump_cargo_version.sh:105]: nit: Should this say `skipping` like the other checks?
```suggestion
                echo "!! File current version ${p_version} doesn't match --current-tag ${current_tag}" - skipping
```

**Review by nick-mobilecoin — CHANGES_REQUESTED** [review]
Nice to see the script for bumping
A couple things:
1. It looks like `go-grpc-gateway/testing/Cargo.toml` and `util/cli/Cargo.toml` should be updated. In #2275 `util/cli/Cargo.toml` got updated, I'm guessing the `go-grpc` one should also.
2. The lock files for the enclaves did not get updated.  
    (this part is outside the scope of the PR) I looked at the action and it appears that there was a cache hit, https://github.com/mobilecoinfoundation/mobilecoin/runs/7488341280?check_suite_focus=true, for the `Cache rust build binaries` so the `Build rust hardware projects` got skipped.  Should there be another check somewhere to make sure the lock files happen?  Or is this just a case that there happened to be a 1.2.3 build from somewhere?  
    I also wonder if the lock files should be in the cache key

**Review by sugargoat — APPROVED** [review]

**Comment by jgreat** [comment]
@nick-mobilecoin 

1. Yes, there are a couple of packages that have not been versioned, they were not really critical and are being addressed in the v2 release branch.

2. There were some issues in the dev environment when I first submitted this and let the CD run. It took several tries to get through the dev cluster issues and get a full success. You have to rewind back to the the first run to see the cargo build happen, after that run the packages and binaries were cached.

Here's the first run where the cargo build happened: https://github.com/mobilecoinfoundation/mobilecoin/runs/7482889789?check_suite_focus=true

Note: I did see a weird issue with the github interface where the steps still show "skipped" icons. You can expand the sections to see the details.

The cache hit/miss key is calculated from a hash of all the Cargo.toml files and few other things. You can see the details here: https://github.com/mobilecoinfoundation/mobilecoin/blob/feature/v1.2.3-bump-changelog/.github/actions/mobilecoin-cache-rust-binaries/action.yaml

Lock file:  It did get updated, but I guess I didn't include a step in my script to make that happen. I wonder if some of the VSCode pugins/debug/language server stuff did that for me.

**Comment by jgreat** [comment]
Ah, I see now, I didn't realize there were separate lock files for each enclave.  I'll get those updated.

- consensus/enclave/trusted/Cargo.lock
- fog/ingest/enclave/trusted/Cargo.lock
- fog/ledger/enclave/trusted/Cargo.lock
- fog/view/enclave/trusted/Cargo.lock

**Review by nick-mobilecoin — APPROVED** [review]
LGTM Thanks!

**Pull request merged**